src/Message.js

```diff
@@ -36,8 +37,10 @@ class Message extends React.Component {
         }
         try {
             const profile = await this.context.client.getProfile(this.props.event.sender);
+            const avatarUrl = profile.avatar_url && this.context.client.thumbnailLink(profile.avatar_url, 'scale', 48, 48);
```

Comment:
Bleh, this actually will be bad if we call `/profile` for every message rendered. Sure we cache, but on first render the cache will be empty so we'll spam the server for the same profile. We should probably add an in-flight cache on `Client.getProfile` so we return the same HTTP promise for the same user, so we can at least guarantee only 1 HTTP hit per user (currently it's 1:1 based on the number of messages).